packages/edit-site/src/components/page-templates/index.js

```diff
@@ -279,21 +227,33 @@ export default function PageTemplates() {
 	);
 
 	const onChangeView = useEvent( ( newView ) => {
-		setView( newView );
-		if ( newView.type !== layout ) {
-			history.navigate(
-				addQueryArgs( path, {
-					layout: newView.type,
-				} )
-			);
+		if ( newView.type !== view.type ) {
+			// Find a way to retrigger the routing resolution.
+			history.invalidate();
 		}
+		updateView( newView );
 	} );
 
 	return (
 		<Page
 			className="edit-site-page-templates"
 			title={ __( 'Templates' ) }
-			actions={ <AddNewTemplate /> }
+			actions={
+				<>
+					{ isModified && (
+						<Button
+							__next40pxDefaultSize
+							onClick={ () => {
+								resetToDefault();
+								history.invalidate();
+							} }
+						>
+							{ __( 'Reset view' ) }
+						</Button>
```

Comment:
Small detail but we might want to wrap the buttons in an `HStack` with 8px spacing. Ideally I suppose we'd have an `PageActions` (or similar) utility to manage this in one location. Maybe it can be added to `Page`?

Reply:
Yes, you're right, but I kept this on purpose and added it to `Page` in this PR https://github.com/WordPress/gutenberg/pull/72106